### DISCUSSION TITLE
Switch to Bottom Tab

### DISCUSSION THREAD

**Discussion by salisuabubakar** [grahammendick/navigation#725] [General]
Hi,  Is there a way i can click a button to switch to one of the bottom tabs ?  Am working on developing a hospital app and i would like to do that for some of the buttons on the homepage. Thanks

**Comment by grahammendick** [comment]
Hi, yea, you can control the active tab.
```jsx
const [tab, setTab] = useState(0);
<TabBar tab={tab} onChangeTab={setTab}>
```
Then you can call `setTab(1)`, for example, to change tabs programmatically.

**Reply by salisuabubakar** [reply]
Hi, Thanks for the quick response as always. 
I have two component one is App.js  and the other is Home.js

In my App.js i have this :
      <TabBar
        scrollable={false}
        primary={true}
        bottomTabs={true}
        labelVisibilityMode='labeled'
        activeIndicatorColor="transparent"
        barTintColor={'#FFFFFF'}
        selectedTintColor={"#05A550"}
        unselectedTintColor={"#5C5C5C"}
        rippleColor="#000000"
        tab={tab}
        onChangeTab={setTab}
      >

and in my Home.js i have: 
              <Pressable
                onPress={() => {
               //Onpress to switch to a tab
              >
                <Text>
                  More
                </Text>
              </Pressable>

How can i click the onpress button in home.js to switch a tab. 

Hope this is clear. Thanks

**Reply by grahammendick** [reply]
You can use any of the typical React techniques to communicate the press event from the `Home` component up to the `App` component. For example, you could use React context or a 3rd-party state library. (You might even be able to use event bubbling but I've not tried it in React Native).

**Reply by salisuabubakar** [reply]
Hi,  i used React Context and its working perfectly. Thanks soo much.

**Reply by grahammendick** [reply]
You can also do it by passing an `onPress` prop from the `App` to the `Home` component
```jsx
<NavigationStack>
  <Scene stateKey="home">
    <Home onChangeTabPress={() => setTab(1)} />
  </Scene>
```

**Reply by salisuabubakar** [reply]
Hi, i have it working with React Context but i wish to understand how i can get the props in the home component. Thanks

**Reply by grahammendick** [reply]
```jsx
// In App
<NavigationStack>
  <Scene stateKey="home">
    <Home onChangeTabPress={() => setTab(1)} />
  </Scene>
</NavigationStack>

// Home component
const Home = ({onChangeTabPress}) => {
  <Pressable onPress={onChangeTabPress}>
}
```

**Reply by salisuabubakar** [reply]
HI, its working perfectly just like React Context example. Please advise, Which is the best solution for instance performace. Thanks

**Reply by grahammendick** [reply]
The best is whatever one you prefer

**Reply by salisuabubakar** [reply]
Ok. I think the prop is simpler with less code. Thanks

**Reply by salisuabubakar** [reply]
Hi, am having some challenges using both react context and the function props. Either of them freezes the app after clicking on the onnPress function. When i click on the function it works but then when i go back to the home.js and click on any button the app freezes. Can you please check. Thanks

**Reply by grahammendick** [reply]
Hey, I did a lot of work [on the `TabBar` in the latest release](https://github.com/grahammendick/navigation/pull/724). Could you update to navigation-react-native v9.50 and give it another go, please?

**Reply by salisuabubakar** [reply]
Hi, i have updated to 9.5.0 and its working smoothly. No freezing now. Thanks. I really dont know how to thank you enough. Thanks soo much.
